### PR TITLE
Support PID logging

### LINKED ISSUE #93: Add CAF::Log option to record the PID
Look at the following snippet from the `ncd.log` file:

```
2015/07/06-14:06:26 [INFO] running component: nss
2015/07/06-14:06:26 ---------------------------------------------------------
2015/07/06-14:06:35
2015/07/06-14:06:35 ------------------------------------------------------------
2015/07/06-14:06:35 [INFO] NCM-NCD version 1.4.1 started by root at: Mon Jul  6 14:06:35 2015
2015/07/06-14:06:39 [WARN] signal handler: received signal: INT
2015/07/06-14:06:39 [ERROR] ncd exiting gracefully after signal hit.
2015/07/06-14:09:52
2015/07/06-14:09:52 =========================================================
2015/07/06-14:09:52
2015/07/06-14:09:52 [OK]   0 errors, 0 warnings executing configure
```

With a little bit of guesswork and intuition, you can figure out what’s going on.  But it would be easier to read if the PID of the `ncm-ncd` process was included, like this:

```
[1273] 2015/07/06-14:06:26 [INFO] running component: nss
[1273] 2015/07/06-14:06:26 ---------------------------------------------------------
[1280] 2015/07/06-14:06:35
[1280] 2015/07/06-14:06:35 ------------------------------------------------------------
[1280] 2015/07/06-14:06:35 [INFO] NCM-NCD version 1.4.1 started by root at: Mon Jul  6 14:06:35 2015
[1280] 2015/07/06-14:06:39 [WARN] signal handler: received signal: INT
[1280] 2015/07/06-14:06:39 [ERROR] ncd exiting gracefully after signal hit.
[1273] 2015/07/06-14:09:52
[1273] 2015/07/06-14:09:52 =========================================================
[1273] 2015/07/06-14:09:52
[1273] 2015/07/06-14:09:52 [OK]   0 errors, 0 warnings executing configure
```

This example occurs when the user manually launches `ncm-ncd` shortly after the dispatcher decides there is work to do.

Can we add an option to `CAF::Log` so the PID can be recorded, and then modify `ncm-ncd` to use that option when setting up the logger for `ncd.log`?

### PR DESCRIPTION
**Pull request by stdweird**
Fixes #93
Based on #126
Adding support to ncm-ncd is handled in https://github.com/quattor/ncm-ncd/issues/55

### CODE REVIEW

File: src/main/perl/Log.pm [hunk 1]
```diff
@@ -18,6 +18,8 @@ use Readonly;
 Readonly our $FH => 'FH';
 Readonly our $FILENAME => 'FILENAME';
 Readonly my $TSTAMP => 'TSTAMP';
+# $PID exists 'use English;' (it's $$)
```
Comment (jrha) [inline src/main/perl/Log.pm]: Huh?

File: src/main/perl/ObjectText.pm [hunk 2]
```diff
@@ -0,0 +1,293 @@
...
+    my $fh = $sc->filewriter('/some/path');
+    if (!defined($fh)) {
+        $self->error("Failed to retrieve filewriter: $sc->{fail}");
+        return;
+    }
+    $fh->close();
+
+=head1 DESCRIPTION
+
+This class simplyfies text handling via stringification and producing
```
Comment (jrha) [inline src/main/perl/ObjectText.pm]: simplifies

File: src/main/perl/ObjectText.pm [hunk 3]
```diff
@@ -0,0 +1,293 @@
...
+The text is added to the filehandle.
+It's up to the consumer to cancel
+and/or close the instance.
+
+All C<CAF::FileWriter> initialisation options are supported
+and passed on. (If no C<log> option is provided,
+ the one from the current instance is passed).
+
+Two new options C<header> and C<footer> are supported
+ to resp. prepend and append to the text.
```
Comment (jrha) [inline src/main/perl/ObjectText.pm]: respectively?

File: src/test/perl/modules/testapp.pm [hunk 4]
```diff
@@ -14,13 +14,40 @@ use strict;
...
 our @ISA = qw(CAF::Application);
 
+sub app_options {
+
+  push(my @array,
+       {
+           NAME =>'logfile=s',
+           HELP =>'log path/filename to use',
+       },
+      );
```
Comment (jrha) [inline src/test/perl/modules/testapp.pm]:
```perl
  push(my @array, {
    NAME =>'logfile=s',
    HELP =>'log path/filename to use',
  });
```

File: src/test/perl/modules/testapp.pm [hunk 5]
```diff
@@ -14,13 +14,40 @@ use strict;
...
+
+sub _initialize {
+    my $self = shift;
+
+    $self->{'LOG_TSTAMP'} = 1;
+    $self->{'LOG_PROCID'} = 1;
+
+    # start initialization of CAF::Application
+    if($self->SUPER::_initialize(@_)) {
+        # why is this not the default?
```
Comment (jrha) [inline src/test/perl/modules/testapp.pm]: why indeed?

File: src/test/perl/object.t [hunk 6]
```diff
@@ -28,6 +29,14 @@ Test all methods for C<CAF::Object>
 
 =over
 
+=item exports
+
+=cut
+
+is(SUCCESS, LC::Exception::SUCCESS, 'CAF::Object exported SUCCESS is LC::Excpetion::SUCCESS');
```
Comment (jrha) [inline src/test/perl/object.t]: LC::Ex**cep**tion::SUCCESS

File: src/test/perl/objecttext.t [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+$fh = $ot->filewriter("/some/name");
+isa_ok($fh, "CAF::FileWriter", "CAF::ObjectText filewriter method returns CAF::FileWriter");
+is("$fh", $TEXTEOL, "File contents as expected");
+
+my $header = "HEADER"; # no newline, eol should add one
+my $footer = "FOOTER"; # no newline, eol should add one
+$fh = $ot->filewriter("/some/name",
+               header => $header,
+               footer => $footer,
+               );
```
Comment (jrha) [inline src/test/perl/objecttext.t]:
```perl
$fh = $ot->filewriter(
    "/some/name",
    header => $header,
    footer => $footer,
);
```

**Comment by jrha** [comment]
Sorry, realised too late that some of the commits are in other PRs.

File: src/main/perl/Log.pm (hunk 1 above)
Comment (stdweird) [inline src/main/perl/Log.pm]: i clarified the comment to 
`# variable $PID exists when using 'use English;' (and it is $$)`
but it's for people wonderng why i'm not using `$PID` for this readonly.

**Comment by stdweird** [comment]
@jrha comments addressed

**Comment by jrha** [comment]
LGTM, @ned21?

**Pull request merged**